openest/test/generate/test_functions.py

```diff
@@ -1,66 +1,69 @@
-import datetime
 import numpy as np
+import pandas as pd
 import xarray as xr
+
 from openest.generate.base import Constant
 from openest.generate.daily import YearlyDayBins
 from openest.generate.functions import Scale, Instabase, SpanInstabase
 from test_daily import test_model
 
-def make_year_ds(year, values):
-    return xr.Dataset({'x': (['time'], values)},
-                      coords={'time': map(lambda nn: datetime.date(year, 1, 1) + datetime.timedelta(nn), np.arange(365))})
 
-def test_Scale():
+def make_year_da(year, values):
+    year = str(year)
+    time_coord = pd.date_range(year + '-01-01', periods=365)
+    return xr.DataArray(values, coords=[('time', time_coord)])
+
+
+def test_scale():
     application = Scale(Constant(33, 'widgets'), {'test': 1. / 11}, 'widgets', 'subwigs').test()
-    for yearresult in application.push(make_year_ds(2000, np.random.rand(365))):
+    for yearresult in application.push(make_year_da(2000, np.random.rand(365))):
         np.testing.assert_equal(yearresult[1], 3)
 
     def check_units():
-        application = Scale(Constant(33, 'widgets'), {'test': 1.8}, 'deg. C', 'deg F.').test()
-        for yearresult in application.push(make_year_ds(2000, np.random.rand(365))):
+        app = Scale(Constant(33, 'widgets'), {'test': 1.8}, 'deg. C', 'deg F.').test()
+        for _ in app.push(make_year_da(2000, np.random.rand(365))):
```

Comment:
I think this should be a `ds`. Why do you say `DataArray`?

Reply:
See my no. 2 point in this comment https://github.com/ClimateImpactLab/open-estimate/pull/15#issuecomment-546094819.

In short: Later in the code you try to take the shape of this input `Dataset`. You can't take the shape of a `Dataset` instance only a `DataArray` (which the `Dataset` contains).

Reply:
We push Datasets, so this won't work upstream. But it won't be a problem since we're mothballing `YearlyDayBins`.  The other place where the `temps.shape` construct  is used in another class in that file is where we first call `weather_change`, which I have used to extract the particular DataArray. There we should improve the variable naming so that it's clear that the output of `weather_change` is not the same class as the input.